Lombiq.Tests.UI/Models/ElasticsearchRunningContext.cs

```diff
@@ -33,8 +36,68 @@ public Task BeforeTestAsync(UITestContext context) =>
 
             (await client.Indices.FlushAsync(index, ct: cancellation)).ThrowIfFailed($"flush index \"{index}\"");
             (await client.Indices.RefreshAsync(index, ct: cancellation)).ThrowIfFailed($"refresh index \"{index}\"");
+
+            // Elasticserch indexing sometimes takes longer, and the testing starts before it finishes. To prevent that,
+            // we are checking if all of the indexing tasks are finished.
+            var elasticIndexManager = provider.GetRequiredService<ElasticIndexManager>();
+            var settingsService = provider.GetRequiredService<ElasticIndexSettingsService>();
+            var indexingTaskManager = provider.GetRequiredService<IIndexingTaskManager>();
+            var indexSettings = await settingsService.GetSettingsAsync();
+            var exactIndexName = indexSettings.FirstOrDefault()?.IndexName;
+
+            var startIndex = 0;
+            const int batchSize = 1000;
+            long lastTaskId = 0;
+            var taskCount = 0;
+
+            // We are getting the last indexing task (regardless of the state). This function works like a cursor, so
+            // there is no way to get directly the last task in the list. Since we have to give a "count" parameter, we
+            // are retrieving the indexing tasks by batches of 1000. Then if there is no more, we get the last one.
+            if (exactIndexName != null)
+            {
+                do
+                {
+                    var tasks = (await indexingTaskManager.GetIndexingTasksAsync(startIndex, batchSize)).ToList();
+                    taskCount = tasks.Count;
+
+                    if (taskCount > 0)
+                    {
+                        lastTaskId = tasks[^1].Id;
+                        startIndex += batchSize;
+                    }
+                }
+                while (taskCount > 0);
```

Comment:
This is a kind of batching iteration, instead of `do... while` maybe
`for (var startIndex = 0; hasTask; startIndex += batchSize)` would be more appropriate.